tests/ignite/metrics/nlp/test_bleu.py

```diff
@@ -28,6 +29,32 @@ def test_wrong_inputs():
     with pytest.raises(NotComputableError):
         Bleu().compute()
 
+    with pytest.raises(ValueError, match='Average must be either "macro" or "micro"'):
+        Bleu(average="macros")
+
+
+@pytest.mark.parametrize(
+    "candidate, references",
+    [
+        ([["a"], ["a"]]),
+        ([["a", "a", "a", "b", "c"]], [[["a", "b", "c"], ["a", "a", "d"]]]),
+        corpus.sample_1,
+        corpus.sample_2,
+        corpus.sample_3,
+        corpus.sample_4,
+    ],
+)
+def test_sentence_bleu(candidate, references):
+    for i in range(1, 8):
```

Comment:
You could refactor to reduce the redundant code introducing a `_test` function
```python
def _test(candidate, references, average):
    for i in range(1, 8):
        weights = tuple([1 / i] * i)
        with warnings.catch_warnings():
            warnings.simplefilter("ignore")
            if average == "macro":
                reference = sentence_bleu(references[0], candidate[0], weights=weights)
            else:
                reference = corpus_bleu(references[0], candidate[0], weights=weights)
        bleu = Bleu(ngram=i, average=average)
        if average == "macro":
            assert pytest.approx(reference) == bleu._sentence_bleu(references[0], candidate[0])
        else:
            assert pytest.approx(reference) == bleu._corpus_bleu(references[0], candidate[0])
        bleu.update((candidate, references))
        assert pytest.approx(reference) == bleu.compute()

@pytest.mark.parametrize(*parametrize_arguments)
def test_macro_bleu(candidate, references):
    _test(candidate, references, average="macro")

@pytest.mark.parametrize(*parametrize_arguments)
def test_micro_bleu(candidate, references):
    _test(candidate, references, average="micro")
```

Moreover, smoothing should be added.